lib/saint.js

```diff
@@ -79,15 +79,23 @@ p.getJobSegment = function(fileId, pageNum, callback){
 	});
 }
 
-p.importJob = function(jobDescription, jobData, callback){
-	var self = this,
-			populateData =[];
-	while(jobData.length){
-		populateData.push(jobData.splice(0,10000));
+p.importJob = function(jobDescription, jobData, callback, jobId){
+	var self = this;
+	var populateData = spliceJobData(jobData);
+	function spliceJobData(jobData) {
+		var splicedJobData = [];
+		while(jobData.length){
+			splicedJobData.push(jobData.splice(0,10000));
+		}
+		return splicedJobData;
 	}
-
 	self.logger("info", "Creating Job");
-	this.importCreateJob(jobDescription, function(err, data){
+	if (typeof jobId === 'undefined') {
+		this.importCreateJob(jobDescription, fnCallImportPopulateJob);
+	} else {
+		fnCallImportPopulateJob(null, jobId);
+	}
+	function fnCallImportPopulateJob(err, data){
 		if(err){ callback(new Error(err.message)); return; };
 		var jobId = data;
```

Comment:
This line could be removed by just avoiding generic arguments in fnCallImportPopulateJob